### DISCUSSION TITLE
output type handlers when fetching oracle objects

### DISCUSSION THREAD

**Discussion by leo-b** [oracle/python-oracledb#88] [Q&A]
I'd like to use Advanced Queuing with object type queues for implementing publish/subscribe queues in Oracle.

The object type returned by dequeue is defined as:
```
CREATE type payment_event as object (
  transaction_tid NUMBER(12,0),
  payment_type CHAR(1),
  payment_tid NUMBER(12,0),
  app VARCHAR2(2)
);
```

Unfortunately I couldn't figure out how to specify oracle to python data type conversion for the `NUMBER(12,0)` attributes. They are currently returned as float while I'd rather need `int` or `Decimal`:

```
>>> m.payload.type  
<oracledb.DbObjectType BACH.PAYMENT_EVENT>
>>> m.payload.type.attributes
[<oracledb.DbObjectAttr TRANSACTION_TID>, <oracledb.DbObjectAttr PAYMENT_TYPE>, <oracledb.DbObjectAttr PAYMENT_TID>, <oracledb.DbObjectAttr APP>]
>>> tt = m.payload.type.attributes[0]
>>> tt.name, tt.type
('TRANSACTION_TID', <DbType DB_TYPE_NUMBER>)
>>> m.payload.TRANSACTION_TID
111.0
>>> type(m.payload.TRANSACTION_TID)
<class 'float'>
```

Is there a way to force a type conversion to `int`?

**Comment by anthony-tuininga** [comment]
Apologies for the lengthy delay! There is no way to do this today. The object support could be enhanced to look at the scale and precision and do the same thing that happens when fetching numbers. If that is of interest to you, please add an enhancement request.

**Reply by leo-b** [reply]
Thanks for clarification. I've added an [enhancement request](https://github.com/oracle/python-oracledb/issues/99).
